### PR TITLE
Courseform Module Tag List and Search Select

### PR DESCRIPTION
**Pull request by tobias-wiese**
# Description
- This is a PR on a feature branch

## Changes in this PR
- adapt the module and exreg selection for courses with the newly created search-select and a tag-list (see screenshot)

## Type of change (remove all that don't apply)
- Refactoring

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Manually tested all use cases this could alter
- could not run E2E test, as no backend is available for it

# Checklist: (remove all that don't apply)

- [x] I added corresponding E2E tests (especially for bugfixes)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new linting warnings or console warnings

# Note: 
I did not update the changelog as this is a refactoring branch for a WIP feature

# Screenshot
grafik

### CODE REVIEW

File: src/components/common/Tag.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <template>
     <span class="mr-2 text-sm font-medium py-1 px-2 rounded align-middle" :class="[bgColor, computedBgColor, textColor]">
         {{ title }}
-        <i class="fa fa-times pl-1" @click="remove" />
+        <i id="removeTag" class="fa fa-times pl- cursor-pointer" @click="remove" />
```
Comment (theovier) [inline src/components/common/Tag.vue:4]: This causes duplicated ids.

Comment (theovier) [inline src/components/common/Tag.vue:4]: I would leave the padding as it makes it easier to click the cross. Also most sites like github do not use the cursor-pointer for their tag system.
```suggestion
        <i class="fa fa-times pl-1" @click="remove" />
```

File: tests/e2e/adminCourseManagement.spec.ts [hunk 2]
```diff
@@ -198,8 +198,9 @@ describe("Course creation, edition and deletion", () => {
     });
 
     it("Can edit modules", () => {
-        cy.get(`input[id=check_module_${course.moduleIds[0]}]`).uncheck();
-        cy.get(`input[id=check_module_${newModule}]`).check();
+        cy.get("span").contains(course.moduleIds[0]).get("i[id='remove']").click();
```
Comment (theovier) [inline tests/e2e/adminCourseManagement.spec.ts:201]: if you choose to remove the non-unique id on the tags, then use a class to identify the correct element

**Review by theovier — CHANGES_REQUESTED** [review]

File: tests/e2e/adminCourseManagement.spec.ts (hunk 2 above)
Comment (tobias-wiese) [reply tests/e2e/adminCourseManagement.spec.ts]: done

File: src/components/common/Tag.vue (hunk 1 above)
Comment (tobias-wiese) [reply src/components/common/Tag.vue]: done

File: src/components/common/Tag.vue [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <template>
     <span class="mr-2 text-sm font-medium py-1 px-2 rounded align-middle" :class="[bgColor, computedBgColor, textColor]">
         {{ title }}
-        <i id="removeTag" class="fa fa-times pl- cursor-pointer" @click="remove" />
+        <i class="fa fa-times cursor-pointer remove-tag" @click="remove" />
```
Comment (theovier) [inline src/components/common/Tag.vue:4]:
```suggestion
        <i class="fa fa-times pl-1 cursor-pointer remove-tag" @click="remove" />
```

**Review by theovier — APPROVED** [review]

File: tests/e2e/adminCourseManagement.spec.ts (hunk 2 above)
Comment (tobias-wiese) [reply tests/e2e/adminCourseManagement.spec.ts]: done

**Pull request merged**